### PR TITLE
ENH: Expose transform

### PR DESCRIPTION
**Pull request by JohnMoutafis**
Exposes Shapely's `transform` function according to ENH: #2010

### CODE REVIEW

File: geopandas/base.py [hunk 1]
```diff
@@ -1260,6 +1260,45 @@ def segmentize(self, max_segment_length):
         """
         return _delegate_geo_method("segmentize", self, max_segment_length)
 
+    def transform(self, transformation, include_z=False):
+        """Returns a ``GeoSeries`` with the transformation function
+        applied to the geom column coordinates.
```
Comment (martinfleis) [inline geopandas/base.py:1265]:
```suggestion
        applied to the geometry coordinates.
```

File: geopandas/base.py [hunk 2]
```diff
@@ -1260,6 +1260,45 @@ def segmentize(self, max_segment_length):
...
+
+        Parameters
+        ----------
+        transformation : Callable
+            A function that transforms a (N, 2) or (N, 3) ndarray of float64
+            to another (N,2) or (N, 3) ndarray of float64
+        include_z : bool, default False
+            If True include the third dimension in the coordinates array that
+            is passed to the transformation function. If a geometry has no third
```
Comment (martinfleis) [inline geopandas/base.py:1274]:
```suggestion
            is passed to the ``transformation`` function. If a geometry has no third
```

File: geopandas/base.py [hunk 3]
```diff
@@ -1260,6 +1260,45 @@ def segmentize(self, max_segment_length):
...
+        >>> s.transform(lambda x: x + 1)
+        0    POINT (1.00000 1.00000)
+        dtype: geometry
+
+        >>> s = geopandas.GeoSeries([Polygon([(0, 0), (1, 1), (0, 1)])])
+        >>> s.transform(lambda x: x * [2, 3])
+        0    POLYGON ((0.00000 0.00000, 2.00000 3.00000, 0.00000 3.00000...
+        dtype: geometry
+
+        By default the third dimension is ignored
```
Comment (martinfleis) [inline geopandas/base.py:1294]:
```suggestion
        By default the third dimension is ignored and you need explicitly include it:
```

File: geopandas/tests/test_geom_methods.py [hunk 4]
```diff
@@ -915,6 +912,33 @@ def test_segmentize_linestrings(self):
         assert_geoseries_equal(expected_g1, result_g1)
         assert_geoseries_equal(expected_g5, result_g5)
 
+    def test_transform(self):
```
Comment (martinfleis) [inline geopandas/tests/test_geom_methods.py:915]: Can you also include a case with 3D geoms but `include_z=False`?

**Review by martinfleis — COMMENTED** [review]
Thanks, looking great! I've left a few minor suggestions.

Can you also add it to `doc/source/docs/reference/geoseries.rst` to be included in docs and add a note to the changelog.md?

**Comment by JohnMoutafis** [comment]
@martinfleis I have updated my branch according to your suggestions!

**Review by m-richards — APPROVED** [review]
Thanks @JohnMoutafis, looking good to me. Will wait and see if Martin has anything further to add before merging.

**Review by martinfleis — APPROVED** [review]
Thanks!

**Pull request merged**